File: prisioner/java/src/main/java/com/everis/competition/prisoner/Decision.java
package com.everis.competition.prisoner;

import java.util.InputMismatchException;

public class Decision {
	
	
	public static final int NEGATION = 0;
	public static final int CONFESS = 1;
	public static final int BRIBERY = 2;
	
    private int myDecision;
    private int otherPrisonerDecision;
     
    
    public int getPacoDecision() {
        return myDecision;
    }

    public int getAntonDecision() {
        return otherPrisonerDecision;
    }

    public Decision(int one, int theOther) {
    	if((one> BRIBERY) || (one < NEGATION)) {
    		// Sanity check: Consider to add an exception
    		one = NEGATION;
    	}
    	if((theOther> BRIBERY) || (theOther < NEGATION)) {
    		// Sanity check: Consider to add an exception
    		theOther = NEGATION;
    	}
        this.myDecision = one;
        this.otherPrisonerDecision = theOther;
    }
}

File: prisioner/js/prisoner/Decision.js
/**
 * http://usejsdoc.org/
 */
const DECISION_OPTIONS = {
		NEGATION: 0,
		CONFESS: 1,
		BRIBERY:2
}


var Decision = function(one, theOther){
	if((one>DECISION_OPTIONS.BRIBERY || one<DECISION_OPTIONS.NEGATION)){
		one = DECISION_OPTIONS.NEGATION;
	}
	if((theOther>DECISION_OPTIONS.BRIBERY || theOther<DECISION_OPTIONS.NEGATION)){
		theOther = DECISION_OPTIONS.NEGATION;
	}
	this._myDecision = one;
	this._otherPrisonerDecision = theOther;
}

Decision.prototype.getPacoDecision = function(){
	return this._myDecision;
}

Decision.prototype.getAntonDecision = function(){
	return this._otherPrisonerDecision;
}

module.exports.DECISION_OPTIONS = DECISION_OPTIONS;
module.exports.Decision = Decision;